packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,349 @@
+
+  /// Test whether the `point` (given in global coordinates) lies within this
+  /// component. The top and the left borders of the component are inclusive,
+  /// while the bottom and the right borders are exclusive.
+  @override
+  bool containsPoint(Vector2 point) {
+    final local = absoluteToLocal(point);
+    return (local.x >= 0) &&
+        (local.y >= 0) &&
+        (local.x < _size.x) &&
+        (local.y < _size.y);
   }
 
-  /// Set the top left position regardless of the anchor
-  set topLeftPosition(Vector2 position) {
-    this.position = position + (anchor.toVector2()..multiply(scaledSize));
+  /// Convert local coordinates of a point [point] inside the component
+  /// into the parent's coordinate space.
+  Vector2 positionOf(Vector2 point) {
+    return _transform.localToGlobal(point);
   }
 
-  /// Get the absolute top left position regardless of whether it is a child or not
-  Vector2 get absoluteTopLeftPosition {
-    final p = parent;
-    if (p is PositionComponent) {
-      return p.absoluteTopLeftPosition + topLeftPosition;
-    } else {
-      return topLeftPosition;
+  /// Similar to [positionOf()], but applies to any anchor point within
+  /// the component.
+  Vector2 positionOfAnchor(Anchor anchor) {
+    if (anchor == _anchor) {
+      return position;
     }
+    return positionOf(Vector2(anchor.x * size.x, anchor.y * size.y));
   }
 
-  /// Get the position that everything in this component is positioned in relation to
-  /// If this component has no parent the absolute parent position is the origin,
-  /// otherwise it's the parents absolute top left position
-  Vector2 get absoluteParentPosition {
-    final p = parent;
-    if (p is PositionComponent) {
-      return p.absoluteTopLeftPosition;
-    } else {
-      return Vector2.zero();
+  /// Convert local coordinates of a point [point] inside the component
+  /// into the global (world) coordinate space.
+  Vector2 absolutePositionOf(Vector2 point) {
+    var parentPoint = positionOf(point);
+    var ancestor = parent;
+    while (ancestor != null) {
+      if (ancestor is PositionComponent) {
+        parentPoint = ancestor.positionOf(parentPoint);
+      }
+      ancestor = ancestor.parent;
     }
+    return parentPoint;
   }
 
-  /// Get the position of the center of the component's bounding rectangle
-  Vector2 get center {
-    if (anchor == Anchor.center) {
-      return position;
-    } else {
-      return anchor.toOtherAnchorPosition(position, Anchor.center, scaledSize)
-        ..rotate(angle, center: absolutePosition);
+  /// Similar to [absolutePositionOf()], but applies to any anchor
+  /// point within the component.
+  Vector2 absolutePositionOfAnchor(Anchor anchor) {
+    var point = positionOfAnchor(anchor);
+    var ancestor = parent;
+    while (ancestor != null) {
+      if (ancestor is PositionComponent) {
+        point = ancestor.positionOf(point);
+      }
+      ancestor = ancestor.parent;
+    }
+    return point;
```

Comment:
could extract an `absolutePositon` method here to be used twice?

Reply:
done